core/eolearn/core/eodata_io.py

```diff
@@ -453,6 +454,14 @@ def _write_to_file(cls, data: _T, file: Union[BinaryIO, gzip.GzipFile], path: st
         file.write(json_data.encode())
 
 
+class FeatureIOTimeStamp(FeatureIOJson[_T]):
```

Comment:
This FeatureIO is specialized for `List[dt.datetime]` and should be marked as such. Just change both occurrences of `_T` with the mentioned type. This will correctly adjust the types of `save` and `load` methods.

Also we treat timestamp as a single word, therefore you should use `Timestamp` and not `TimeStamp`.